src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -225,20 +231,26 @@ export class FocusMonitor implements OnDestroy {
 
     const nativeElement = coerceElement(element);
 
+    // If the element is inside the shadow DOM, we need to bind our focus/blur listeners to
+    // the shadow root, rather than the `document`, because the browser won't emit focus events
+    // to the `document`, if focus is moving within the same shadow root.
+    const rootNode = (_getShadowRoot(nativeElement) as HTMLElement|null) || this._getDocument();
+
     // Check if we're already monitoring this element.
     if (this._elementInfo.has(nativeElement)) {
-      const cachedInfo = this._elementInfo.get(nativeElement);
-      cachedInfo!.checkChildren = checkChildren;
-      return cachedInfo!.subject.asObservable();
+      const cachedInfo = this._elementInfo.get(nativeElement)!;
+      cachedInfo.checkChildren = checkChildren;
```

Comment:
I see that you didn't change this, but is this right? Won't this cause all previous calls to monitor on this element to behave according to the `checkChildren` parameter on the current call?

Reply:
Judging by how it's written, I think that's the point. E.g. you could turn a shallow watch into a deep one by calling `monitor` again with `checkChildren`. We could separate it out and have 2 separate streams, but we'd have to make a cache key out of the options.

Reply:
Yeah, I'm not sure it should work that way. The main way I see it being called twice is if there's e.g. a directive and a component that need to know about focus state on the same element. They might have different requirements about whether to include the children, and we wouldn't want them to affect each other's subscriptions.

So it seems like a bug, but a pre-existing one and I haven't seen anyone complain about it yet, so I think its safe to leave this for another PR.